core/blockchain.go

```diff
@@ -185,6 +185,20 @@ func DefaultCacheConfigWithScheme(scheme string) *CacheConfig {
 	return &config
 }
 
+// BlockchainLogger is used to collect traces during chain processing.
+// Please make a copy of the referenced types if you intend to retain them.
+type BlockchainLogger interface {
+	vm.EVMLogger
+	state.StateLogger
+	// OnBlockStart is called before executing `block`.
+	// `td` is the total difficulty prior to `block`.
+	OnBlockStart(block *types.Block, td *big.Int, finalized *types.Header, safe *types.Header, chainConfig *params.ChainConfig)
+	OnBlockEnd(err error)
+	OnGenesisBlock(genesis *types.Block, alloc GenesisAlloc)
+	OnBeaconBlockRootStart(root common.Hash)
+	OnBeaconBlockRootEnd()
+}
```

Comment:
I'd prefer one super interface, maybe `ChainLogger`, and have `Blockchainlogger` contain only the block mechanics.